partiql-planner/src/testFixtures/resources/inputs/basics/case.sql

```diff
@@ -1,54 +1,309 @@
---#[case-00]
+-- -----------------------------
+--  Exact Numeric
+-- -----------------------------
+
+--#[case-when-00]
+-- type: (int32)
+CASE t_item.t_bool
+    WHEN true THEN 0
+    WHEN false THEN 1
+    ELSE 2
+END;
+
+--#[case-when-02]
+-- type: (int32)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16  -- cast(.. AS INT4)
+    ELSE t_item.t_int32           -- INT4
+END;
+
+--#[case-when-03]
+-- type: (int64)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16   -- cast(.. AS INT8)
+    WHEN 'b' THEN t_item.t_int32   -- cast(.. AS INT8)
+    ELSE t_item.t_int64            -- INT8
+END;
+
+--#[case-when-04]
+-- type: (int)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16  -- cast(.. AS INT)
+    WHEN 'b' THEN t_item.t_int32  -- cast(.. AS INT)
+    WHEN 'c' THEN t_item.t_int64  -- cast(.. AS INT)
+    ELSE t_item.t_int             -- INT
+END;
+
+--#[case-when-05]
+-- type: (int)
+CASE t_item.t_string
+    WHEN 'b' THEN t_item.t_int32  -- cast(.. AS INT)
+    WHEN 'c' THEN t_item.t_int64  -- cast(.. AS INT)
+    ELSE t_item.t_int             -- INT
+END;
+
+--#[case-when-06]
+-- type: (int)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16  -- cast(.. AS INT)
+    WHEN 'b' THEN t_item.t_int32  -- cast(.. AS INT)
+    ELSE t_item.t_int             -- INT
+END;
+
+--#[case-when-07]
+-- type: (int64)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int32  -- cast(.. AS INT8)
+    WHEN 'b' THEN t_item.t_int64  -- INT8
+    ELSE t_item.t_int16           -- cast(.. AS INT8)
+END;
+
+--#[case-when-08]
+-- type: (int|null)
+-- nullable default
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16  -- cast(.. AS INT)
+    WHEN 'b' THEN t_item.t_int32  -- cast(.. AS INT)
+    ELSE t_item.t_int_null        -- INT
+END;
+
+--#[case-when-09]
+-- type: (int|null)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_int16_null -- cast(.. AS INT)
+    WHEN 'b' THEN t_item.t_int32      -- cast(.. AS INT)
+    ELSE t_item.t_int
+END;
+
+--#[case-when-10]
+-- type: (decimal|null)
+-- nullable branch
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal
+    WHEN 'b' THEN t_item.t_int32
+    ELSE NULL
+END;
+
+--#[case-when-11]
+-- type: (int|null|missing)
+-- TODO should really be (int|missing) but our translation of coalesce doesn't consider types.
+COALESCE(CAST(t_item.t_string AS INT), 1);
+
+-- -----------------------------
+--  Approximate Numeric
+-- -----------------------------
+
+-- TODO model approximate numeric
+-- We do not have the appropriate StaticType for this.
+
+--#[case-when-12]
+-- type: (float64|null)
```

Comment:
shouldn't have `null` in comment. test in `PlanTyperTestsPorted` looks to be correct

```suggestion
-- type: (float64)
```